### PR TITLE
Added User model, DTOs, repository, service, and AuthController

### PR DESCRIPTION
**Pull request by vikkushnir**
Added User model, DTOs, repository, service, and AuthController. Updated swagger

### CODE REVIEW

File: src/main/java/online/book/store/validation/FieldMatch.java [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+@Target({ElementType.TYPE, ElementType.ANNOTATION_TYPE})
+@Retention(RetentionPolicy.RUNTIME)
+@Constraint(validatedBy = FieldMatchValidator.class)
+public @interface FieldMatch {
+    String message() default "Fields must match";
+    Class<?>[] groups() default {};
+    Class<? extends Payload>[] payload() default {};
+    String first();
+    String second();
```
Comment (andrii-hoienko) [inline src/main/java/online/book/store/validation/FieldMatch.java]: This fields are not used why do you need them?

File: src/main/java/online/book/store/validation/FieldMatchValidator.java [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+import java.util.Objects;
+import online.book.store.dto.request.user.UserRegistrationRequestDto;
+
+public class FieldMatchValidator
+        implements ConstraintValidator<FieldMatch, UserRegistrationRequestDto> {
+    @Override
+    public boolean isValid(UserRegistrationRequestDto requestDto,
+                           ConstraintValidatorContext constraintValidatorContext) {
+        return Objects.equals(requestDto.getPassword(), requestDto.getRepeatPassword());
+    }
```
Comment (andrii-hoienko) [inline src/main/java/online/book/store/validation/FieldMatchValidator.java]: I believe this annotation should work with 2 any fields

**Review by andrii-hoienko — CHANGES_REQUESTED** [review]
Nice, left some questions and comments

File: src/main/java/online/book/store/validation/FieldMatchValidator.java [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+        Object fieldValue = new BeanWrapperImpl(value)
+                .getPropertyValue(field);
+        Object fieldMatchValue = new BeanWrapperImpl(value)
+                .getPropertyValue(fieldMatch);
+
+        if (fieldValue != null) {
+            return fieldValue.equals(fieldMatchValue);
+        } else {
+            return fieldMatchValue == null;
+        }
```
Comment (karma-o) [inline src/main/java/online/book/store/validation/FieldMatchValidator.java]:
```suggestion
        return Objects.equals(fieldValue, fieldMathcValue);
```

File: src/main/java/online/book/store/dto/request/user/UserRegistrationRequestDto.java [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+    private String email;
+    @NotBlank(message = "field can't be empty")
+    @Size(min = 8, message = "field should contain at least 8 character")
+    private String password;
+    @NotBlank(message = "please repeat your password!")
+    private String repeatPassword;
+    @NotBlank(message = "field can't be empty")
+    private String firstName;
+    @NotBlank(message = "field can't be empty")
+    private String lastName;
```
Comment (karma-o) [inline src/main/java/online/book/store/dto/request/user/UserRegistrationRequestDto.java]: I do not think we need a custom message, default ones are usually fine, and we do not need to complex things

**Review by karma-o — APPROVED** [review]
GJ!

**Pull request merged**